app/models/event.rb

```diff
@@ -66,6 +66,17 @@ def search_activities(author = nil, query_string = "", filter = "all")
   end
   alias_method :activities, :search_activities
 
+  def activities_per_day(author = nil, query_string = "", filter = "all")
+    activities = search_activities(author, query_string, filter)
+    activities.sort_by! { |activity| activity.dates.first }
+
+    activities.each_with_object(Hash.new { [] }) do |activity, grouped_by_day|
```

Comment:
I hope it's not too clever. I find default Hash values very useful, but they are not great for readability. I have thought about using [Hash#default_proc=](http://ruby-doc.org/core-2.2.2/Hash.html#method-i-default_proc-3D).